source/jormungandr/jormungandr/street_network/geovelo.py

```diff
@@ -105,11 +105,60 @@ def _make_request_arguments_bike_details(cls, bike_speed_mps):
             'averageSpeed': bike_speed,  # in km/h, BEGINNER sets it to 13
         }
 
+    @classmethod
+    def _keep_obj_for_mode(cls, obj, modes):
+        # This filter is only necessary for tests without any physical_modes.
+        # In real life data physical_modes is never empty with depth = 2
+        if len(obj.stop_point.physical_modes) == 0:
+            return True
+        for mode in obj.stop_point.physical_modes:
+            if mode.uri in modes:
+                return True
+
     @classmethod
     def _make_request_arguments_isochrone(cls, origins, destinations, bike_speed_mps=3.33):
+        train_modes = ['physical_mode:Train', 'physical_mode:RapidTransit']
+        metro_modes = ['physical_mode:Metro']
+        tram_modes = ['physical_mode:Tramway']
+        bus_modes = ['physical_mode:Car', 'physical_mode:Bus']
+
+        origins_coord = set()
+        destinations_coord = set()
+        if len(origins) == 1:
+            origins_coord.update(cls._pt_object_summary_isochrone(o) for o in origins)
+
+            # Filter matrix points with modes having priority order:
+            # Train', 'RapidTransit', 'Metro', 'Tramway', Car, Bus
+            for modes in (train_modes, metro_modes, tram_modes, bus_modes):
+                if len(destinations_coord) < 50:
+                    destinations_coord.update(
+                        cls._pt_object_summary_isochrone(o)
+                        for o in destinations
+                        if cls._keep_obj_for_mode(o, modes)
+                    )
+                else:
+                    break
+
+            destinations_coord = list(destinations_coord)
+            del destinations_coord[50:]
```

Comment:
:warning:  this code may not return what you want... `set()` is an unordered container... `destinations_coord`'s elements are not ordered by appending order( not FIFO ), when you remove elements, it may remove unwanted elements